pybop/transformation/transformations.py

```diff
@@ -109,19 +109,19 @@ def jacobian_S1(self, q: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
 
     def log_jacobian_det(self, q: np.ndarray) -> float:
         """See :meth:`Transformation.log_jacobian_det()`."""
-        return np.sum(np.log(np.abs(self._coefficient)))
+        return np.sum(np.log(np.abs(self.coefficient)))
```

Comment:
This should be more efficient for larger arrays, since we can reduce the number of intermediate array creations – also, slightly more readable!

```suggestion
        return np.log(np.abs(self.coefficient)).sum()
```